services/eventService/eventService.go

```diff
@@ -1,43 +1,250 @@
 package eventService
 
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/patrickmn/go-cache"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
 // An Event represents a HackSoc event.
 type Event struct {
-	Name        string
-	Description string
-	Location    string
-	Date        string
-	URL         string
-	ImageURL    string
-	Upcoming    bool
+	Name            string
+	Description     string
+	Location        string
+	URL             string
+	ImageURL        string
+	AttendingCount  int
+	InterestedCount int
+	StartTime       time.Time
+	EndTime         time.Time
+}
+
+func (e *Event) GetInterestLine() string {
+	if e.StartTime.After(time.Now()) || e.EndTime.After(time.Now()) {
+		totalCount := e.AttendingCount + e.InterestedCount
+		return fmt.Sprintf("%d going or interested", totalCount)
+	}
+	return fmt.Sprintf("%d attended", e.AttendingCount)
+}
+
+func (e *Event) GetShortDate() string {
+	if e.StartTime.Year() == time.Now().Year() && e.StartTime.After(time.Now()) {
+		startTime := e.StartTime.Format("January 2")
+		if e.StartTime.YearDay() != e.EndTime.YearDay() {
+			if e.StartTime.Month() == e.EndTime.Month() {
+				return fmt.Sprintf("%s - %s", startTime, e.EndTime.Format("2"))
+			}
+			var endTime string
+
+			if e.EndTime.Year() == time.Now().Year() {
+				endTime = e.EndTime.Format("January 2")
+			} else {
+				endTime = e.EndTime.Format("January 2, 2006")
+			}
+
+			return fmt.Sprintf("%s - %s", startTime, endTime)
+		} else {
+			return startTime
+		}
+	}
+	return e.StartTime.Format("January 2, 2006")
 }
 
-// GetEvents gets the events.
-func GetEvents() []*Event {
-	events := []*Event{
-		&Event{"GreatUniHack 2016", "A first test", "Kilburn Building", "September 16 at 5:30pm", "https://google.com/", "https://www.google.co.uk/images/branding/product/ico/googleg_lodp.ico", true},
-		&Event{"Test 2", "A second test", "Here", "June 1 at 12:00pm", "https://google.com/", "https://www.google.co.uk/images/branding/product/ico/googleg_lodp.ico", false},
-		&Event{"Test 3", "A third test", "Here", "May 25 at 10:00am", "https://google.com/", "https://www.google.co.uk/images/branding/product/ico/googleg_lodp.ico", false},
+const timeFormatLayout = "2006-01-02T15:04:05-0700"
+
+var accessToken string
+
+var eventsCache *cache.Cache
+
+func init() {
+	if os.Getenv("FB_APP_ID") == "" {
+		log.Println("Environment variable FB_APP_ID is not assigned.")
+		return
 	}
 
-	return events
+	if os.Getenv("FB_SECRET") == "" {
+		log.Println("Environment variable FB_SECRET is not assigned.")
+		return
+	}
```

Comment:
Nit: if I won't have both "FB_APP_ID" and "FB_SECRET", I won't be notified about the latter.

Reply:
Agree, we should show all environment variables we're lacking.

Fixed! :+1: